dvc/stage/run.py

```diff
@@ -12,27 +14,58 @@
 logger = logger.getChild(__name__)
 
 
-def _make_cmd(executable, cmd):
-    if executable is None:
-        return cmd
-    opts = {"zsh": ["--no-rcs"], "bash": ["--noprofile", "--norc"]}
-    name = os.path.basename(executable).lower()
-    return [executable, *opts.get(name, []), "-c", cmd]
-
+def _fish_supports_no_config(executable) -> bool:
+    """
+    Check if the fish shell supports the --no-config option.
 
-def warn_if_fish(executable):
-    if executable is None or os.path.basename(os.path.realpath(executable)) != "fish":
-        return
+    Parameters:
+    executable (str): The path to the fish shell executable.
 
+    Returns:
+    bool: True if the fish version is greater than 3.3.0, False otherwise.
+    """
+    try:
+        output = subprocess.check_output(
+            [executable, "-c", "echo $FISH_VERSION"],  # noqa: S603
+            text=True,
+        )
+    except subprocess.CalledProcessError:
+        logger.warning("could not check fish version, defaulting to False")
+        return False
+    version = Version(output.strip())
+    version_to_check = Version("3.3.0")
+    return version > version_to_check
```

Comment:
so, should it be `>=`?

Reply:
true, changed in https://github.com/iterative/dvc/pull/10445/commits/f4acd67aff23b409716fe69ed434809c900672d8